src/ansible_compat/prerun.py

```diff
@@ -0,0 +1,427 @@
+        "install",
+        "-v",
+    ]
+    if destination:
+        cmd.extend(["-p", destination])
+    cmd.append(f"{collection}")
+
+    _logger.info("Running %s", " ".join(cmd))
+    run = subprocess.run(
+        cmd,
+        universal_newlines=True,
+        check=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+    if run.returncode != 0:
+        _logger.error("Command returned %s code:\n%s", run.returncode, run.stdout)
+        sys.exit(INVALID_PREREQUISITES_RC)
+
+
+@tenacity.retry(  # Retry up to 3 times as galaxy server can return errors
+    reraise=True,
+    wait=tenacity.wait_fixed(30),  # type: ignore
+    stop=tenacity.stop_after_attempt(3),  # type: ignore
+    before_sleep=tenacity.after_log(_logger, logging.WARNING),  # type: ignore
+)
+def install_requirements(requirement: str) -> None:
+    """Install dependencies from a requirements.yml."""
+    if not os.path.exists(requirement):
+        return
+
+    cmd = [
+        "ansible-galaxy",
+        "role",
+        "install",
+        "--roles-path",
+        f"{options.cache_dir}/roles",
+        "-vr",
+        f"{requirement}",
+    ]
+
+    _logger.info("Running %s", " ".join(cmd))
+    run = subprocess.run(
+        cmd,
+        universal_newlines=True,
+        check=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+    if run.returncode != 0:
+        _logger.error(run.stdout)
+        raise RuntimeError(run.returncode)
+
+    # Run galaxy collection install works on v2 requirements.yml
+    if "collections" in yaml_from_file(requirement):
+
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
+            "-p",
+            f"{options.cache_dir}/collections",
+            "-vr",
+            f"{requirement}",
+        ]
+
+        _logger.info("Running %s", " ".join(cmd))
+        run = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        if run.returncode != 0:
+            _logger.error(run.stdout)
+            raise RuntimeError(run.returncode)
+
+
+def prepare_environment(required_collections: Optional[Dict[str, str]] = None) -> None:
+    """Make dependencies available if needed."""
+    if not options.configured:
+        # Allow method to be used without calling the command line, so we can
+        # reuse it in other tools, like molecule.
+        # pylint: disable=import-outside-toplevel,cyclic-import
+        # from ansiblelint.__main__ import initialize_options
+
+        # initialize_options()
+        pass
+
+    if not options.offline:
+        install_requirements("requirements.yml")
+        for req in pathlib.Path(".").glob("molecule/*/requirements.yml"):
+            install_requirements(str(req))
+
+    if required_collections:
+        for name, min_version in required_collections.items():
```

Comment:
Alternatively, just do
```suggestion
        for name, min_version in (required_collections or {}).items():
```